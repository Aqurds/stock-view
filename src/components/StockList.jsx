import { useEffect, useState } from "react";
import { useSelector, useDispatch } from "react-redux";
import { fetchStockSymbols } from "../store/stock/stockSlice";
import Header from "./Header";
import SingleStock from "./SingleStock";
import '../styles/stock-list.css';
import { IoArrowForwardCircleOutline } from 'react-icons/io5';
import { FaSearch } from 'react-icons/fa';

const StockList = () => {
  const stockSymbols = useSelector(state => state.stocks.stocks.slice(11500,11600));
  const stockLoading = useSelector(state => state.stocks.loading);
  const dispatch = useDispatch();
  const [searchQuery, setSearchQuery] = useState('');
  const [filteredStockSymbols, setFilteredStockSymbols] = useState([])

  useEffect(() => {
    dispatch(fetchStockSymbols())
  }, [])

  useEffect(() => {
    const newStockSymbols = stockSymbols.filter((stock) => {
      return stock.includes(searchQuery.trim().toUpperCase())
    })
    setFilteredStockSymbols(newStockSymbols)
  }, [searchQuery])

  return (
    <>
      <Header />
      <div className="search-wrapper">
        <FaSearch className="search-icon"/>
        <input 
          type='text' 
          value={searchQuery}
          onChange={(e) => setSearchQuery(e.target.value)}
          className="search-stock"
          placeholder="Search Stock. Ex: AAMC"
        />
      </div>
      <div className="stock-wrapper">
        { stockLoading && 
        <p className="stock-loading">
          Stock loading ...
        </p> 
        }
        { searchQuery ? (
           filteredStockSymbols && filteredStockSymbols.map((stock) => {
          return (
            <div className="single-stock">
              <a href={`/single-stock/${stock}`} key={stock}>
                <IoArrowForwardCircleOutline className="go-single-stock" />
              </a>
              <a href={`/single-stock/${stock}`} key={stock}>
                {stock}
              </a>
            </div>
          )
        })
        ) : (
          stockSymbols && stockSymbols.map((stock) => {
          return (
            <div className="single-stock">
              <a href={`/single-stock/${stock}`} key={stock}>
                <IoArrowForwardCircleOutline className="go-single-stock" />
              </a>
              <a href={`/single-stock/${stock}`} key={stock}>
                {stock}
              </a>
            </div>
          )
        })
        )}
      </div>
    </>
  )
};

export default StockList;