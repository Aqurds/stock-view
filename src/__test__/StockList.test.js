import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import StockList from '../components/StockList';
import thunk from 'redux-thunk';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('StockList component', () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      stocks: {
        stocks: ['AAPL', 'GOOGL', 'AMZN'],
        loading: false,
      },
    });
  });

  it('renders the component with initial stock symbols', () => {
    const { container } = render(
      <Provider store={store}>
        <StockList />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  // Add more test cases as needed
});
