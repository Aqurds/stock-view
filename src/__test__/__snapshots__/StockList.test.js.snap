// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StockList component renders the component with initial stock symbols 1`] = `
<div>
  <header
    class="header-wrapper"
    data-testid="header"
  >
    <svg
      class="stock-view-logo"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M496 384H64V80c0-8.84-7.16-16-16-16H16C7.16 64 0 71.16 0 80v336c0 17.67 14.33 32 32 32h464c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16zM464 96H345.94c-21.38 0-32.09 25.85-16.97 40.97l32.4 32.4L288 242.75l-73.37-73.37c-12.5-12.5-32.76-12.5-45.25 0l-68.69 68.69c-6.25 6.25-6.25 16.38 0 22.63l22.62 22.62c6.25 6.25 16.38 6.25 22.63 0L192 237.25l73.37 73.37c12.5 12.5 32.76 12.5 45.25 0l96-96 32.4 32.4c15.12 15.12 40.97 4.41 40.97-16.97V112c.01-8.84-7.15-16-15.99-16z"
      />
    </svg>
    <h1
      class="header-text"
    >
      Stock View
    </h1>
    <div
      class="header-icon-wrapper"
    >
      <svg
        class="header-icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5 3a3 3 0 0 1 6 0v5a3 3 0 0 1-6 0V3z"
        />
        <path
          d="M3.5 6.5A.5.5 0 0 1 4 7v1a4 4 0 0 0 8 0V7a.5.5 0 0 1 1 0v1a5 5 0 0 1-4.5 4.975V15h3a.5.5 0 0 1 0 1h-7a.5.5 0 0 1 0-1h3v-2.025A5 5 0 0 1 3 8V7a.5.5 0 0 1 .5-.5z"
        />
      </svg>
      <svg
        class="header-icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 176a80 80 0 1080 80 80.24 80.24 0 00-80-80zm172.72 80a165.53 165.53 0 01-1.64 22.34l48.69 38.12a11.59 11.59 0 012.63 14.78l-46.06 79.52a11.64 11.64 0 01-14.14 4.93l-57.25-23a176.56 176.56 0 01-38.82 22.67l-8.56 60.78a11.93 11.93 0 01-11.51 9.86h-92.12a12 12 0 01-11.51-9.53l-8.56-60.78A169.3 169.3 0 01151.05 393L93.8 416a11.64 11.64 0 01-14.14-4.92L33.6 331.57a11.59 11.59 0 012.63-14.78l48.69-38.12A174.58 174.58 0 0183.28 256a165.53 165.53 0 011.64-22.34l-48.69-38.12a11.59 11.59 0 01-2.63-14.78l46.06-79.52a11.64 11.64 0 0114.14-4.93l57.25 23a176.56 176.56 0 0138.82-22.67l8.56-60.78A11.93 11.93 0 01209.94 26h92.12a12 12 0 0111.51 9.53l8.56 60.78A169.3 169.3 0 01361 119l57.2-23a11.64 11.64 0 0114.14 4.92l46.06 79.52a11.59 11.59 0 01-2.63 14.78l-48.69 38.12a174.58 174.58 0 011.64 22.66z"
        />
      </svg>
    </div>
  </header>
  <div
    class="search-wrapper"
  >
    <svg
      class="search-icon"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
      />
    </svg>
    <input
      class="search-stock"
      placeholder="Search Stock. Ex: AAMC"
      type="text"
      value=""
    />
  </div>
  <div
    class="stock-wrapper"
  />
</div>
`;
